test/integration/h2_fuzz.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "test/common/http/http2/http2_frame.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/integration/h2_capture_fuzz.pb.h"
+#include "test/integration/http_integration.h"
+
+namespace Envoy {
+
+class H2FuzzIntegrationTest : public HttpIntegrationTest {
+public:
+  H2FuzzIntegrationTest(Network::Address::IpVersion version)
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, version) {}
+
+  void initialize() override;
+  void replay(const test::integration::H2CaptureFuzzTestCase&);
+
+private:
+  // Maps between the enum value in h2_capture_fuzz.proto to the correct
+  // function in test/common/http/http2/http2_frame.h
+  // TODO(adip): some methods receive the stream index, which is currently set to 1, this should
+  // be fuzzed further
+  static constexpr uint32_t stream_index_ = 1;
+  static constexpr const char* host_ = "host";
+  static constexpr const char* path_ = "/test/long/url";
```

Comment:
Same might be true about some of these other field, they are potentially fuzzable without running into the H2 binary format issue.